### ISSUE TITLE
DNS Beacon

### ISSUE THREAD

**Issue by ghost** [gloxec/CrossC2#66]
It's not an issue, more a feature request.

I know that CS have DNS implant too, but for windows.
Do you think you could manage to imitate it for linux/mac too?
It would be a big work i think, but at same time it would make CS even better.

Let me know!

**Comment by gloxec** [comment]
Yes, DNS beacon is indeed in the implementation plan list. However, the current focus is still on the realization of the socks proxy and "so/dylib beacon" and its derivative process injection function under the premise of ensuring a small size.
At present, socks proxy, "so/dylib beacon", and process injection have been implemented as singletons, but we can't solve the problem that all functions are integrated in a beacon, which causes its size to be too large. We are still looking for some available event feedback at the bottom of cobaltstrike to complete the automated loading of complex modules.

**Comment by ghost** [comment]
Awesome!
